dart/dynamics/Shape.h

```diff
@@ -191,29 +142,19 @@ class Shape : public virtual common::Subject
   /// Instruct this shape to update its data
   virtual void refreshData();
 
-  /// \brief
-  virtual void draw(renderer::RenderInterface* _ri = nullptr,
-                    const Eigen::Vector4d& _color = Eigen::Vector4d::Ones(),
-                    bool _useDefaultColor = true) const = 0;
-
-  /// Pass in true to prevent this shape from being rendered. Pass in false to
-  /// allow it to render again.
-  void setHidden(bool _hide=true);
+  /// Notify that the color (rgba) of this shape has updated
+  virtual void notifyColorUpdate(const Eigen::Vector4d& color);
```

Comment:
We'll actually have a versioning system that will allow the viewer to know when it needs to update things like color or vertex data.

For the key/value storage, where should that container live? Who should own it? I feel that this should be the job of the end user (DART is not really intended to be a comprehensive rendering engine), and we should design the ShapeFrame so that any customized information that a user might want to attach to their rendering information can be done easily (I would suggest through Addons).